src/table/virtualizedDataTable.jsx

```diff
@@ -1191,214 +1499,89 @@ class VirtualizedDataTable extends Component {
-      headerHeight,
+      groupFarOuterStyle,
+      groupHeaderStyle,
+      groupOuterStyle,
+      groups,
+      headerStyle,
+      rowCountWithEmptySpace,
+      selectionRanges,
+      tableStyle,
+      useDefaultClipboard,
       width,
-    };
-
-    // shim in some default styling
-    headerStyle = _.assign({}, styles.tableHeader, headerStyle || {});
-    gridStyle = _.assign({}, styles.tableGrid, gridStyle || {});
-
-    // ensure we will draw blank rows in the gap between height and the final renderable
-    // row by calculating the number of rows needed to fill the available space and setting
-    // the row count to that value
-    const rowCountWithEmptySpace = Math.max(
-      Math.floor((height - headerHeight) / rowHeight),
-      rowCountUse,
-    );
-
-    this.prvColumnInfo = [];
-    const columns = [];
-    const groups = [];
-    let columnLeft = 0;
-
-    // render all columns, skipping over any ColumnGroups and directly rendering their column
-    // children
-    React.Children.forEach(children, (child) => {
-      if (child) {
-        if (child.type && ((child.type === ColumnGroup) || (child.type.name === 'ColumnGroup'))) {
-          let groupWidth = 0;
-
-          React.Children.forEach(child.props.children, (grandChild) => {
-            if (grandChild) {
-              if (grandChild.type && ((grandChild.type !== Column) && (grandChild.type.name !== 'Column'))) {
-                throw new Error('unexpected child - only Columns can be children of ColumnGroups!');
-              }
-
-              columns.push(this.prvRenderTableColumn({
-                tableProps,
-                styles,
-                column: grandChild,
-                columnLeft,
-              }));
-
-              groupWidth += grandChild.props.width || 0;
-              columnLeft += grandChild.props.width || 0;
-
-              this.prvColumnInfo.push({
-                columnKey: grandChild.props.columnKey,
-                width: grandChild.props.width || 0,
-              });
-            }
-          });
-
-          groups.push({
-            group: child,
-            groupWidth,
-          });
-        } else if (child.type && ((child.type === Column) || (child.type.name === 'Column'))) {
-          columns.push(this.prvRenderTableColumn({
-            tableProps,
-            styles,
-            column: child,
-            columnLeft,
-          }));
-
-          columnLeft += child.props.width || 0;
-
-          this.prvColumnInfo.push({
-            columnKey: child.props.columnKey,
-            width: child.props.width || 0,
-          });
-        } else {
-          throw new Error('unexpected child - only Columns or ColumnGroups can be children of VirtualizedDataTable!');
-        }
-      }
-    });
+    } = this.state;
+    const cellGridHeight = height - (headerHeight + 2); // subtract 2 because of borders
+    const columns = columnInfo.map(columnInfoCur => this.prvRenderTableColumn(columnInfoCur));
```

Comment:
Above we take the set of columns computed in `getDerivedStateFromProps` (formerly right here in `render` as seen above) and render them with prvRenderTableColumn, which was a render function and also required access to the component instance and couldn't be done from a static method